src/test/resources/application.properties

```diff
@@ -1,19 +1,17 @@
-#spring.main.banner-mode=off
+spring.main.banner-mode=off
 logging.level.org.springframework=ERROR
 logging.level.root=INFO
 
 spring.jpa.show-sql=true
 #Do not use this in production
-spring.jpa.hibernate.ddl-auto=create-drop
-spring.cloud.stream.bindings.input.destination=mySource
-spring.datasource.driverClassName=org.postgresql.Driver
-spring.jpa.database-platform=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.driverClassName=org.h2.Driver
```

Comment:
you are running H2 in postgres compatibility mode.  I would leave driver and platform as postgres.  This will require some testing though to confirm it behaves as expected so for this PR leave it be but I would like to see if we can get the tests running using posgres dialect and postgres driver to an H2 database in postgres compatibility mode.

Reply:
If you leave the driver as postgres, it will not recognize the h2 address as valid when trying to give it the `datasource.url`.

Reply:
we would use the postgres URL but the trick is making postgres aware.